diff --git t3_run_tests.sh t3_run_tests.sh
index dda4232..9b23979 100755
--- t3_run_tests.sh
+++ t3_run_tests.sh
@@ -386,6 +386,7 @@ THIS_SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
 cd "$THIS_SCRIPT_DIR" || exit 1
 cd ../../../../../ || exit 1
 CORE_ROOT="${PWD}"
+ACCEPTANCE_ROOT="${CORE_ROOT}/.Build/Web"
 
 # Default variables
 TEST_SUITE="unit"
@@ -393,6 +394,7 @@ DBMS="sqlite"
 DBMS_VERSION=""
 PHP_VERSION="8.2"
 PHP_XDEBUG_ON=0
+PHP_XDEBUG_MODE=""
 PHP_XDEBUG_PORT=9003
 ACCEPTANCE_HEADLESS=1
 ACCEPTANCE_TOPIC="sets"
@@ -422,7 +424,7 @@ OPTIND=1
 # Array for invalid options
 INVALID_OPTIONS=()
 # Simple option parsing based on getopts (! not getopt)
-while getopts ":a:b:s:c:d:i:t:p:xy:nhug" OPT; do
+while getopts ":a:b:s:c:d:i:t:p:xX:y:nhug" OPT; do
     case ${OPT} in
         s)
             TEST_SUITE=${OPTARG}
@@ -466,6 +468,10 @@ while getopts ":a:b:s:c:d:i:t:p:xy:nhug" OPT; do
         x)
             PHP_XDEBUG_ON=1
             ;;
+        X)
+            PHP_XDEBUG_ON=0
+            PHP_XDEBUG_MODE=${OPTARG}
+            ;;
         y)
             PHP_XDEBUG_PORT=${OPTARG}
             ;;
@@ -579,6 +585,10 @@ else
     PHP_FPM_OPTIONS="-d xdebug.mode=debug -d xdebug.start_with_request=yes -d xdebug.client_host=${CONTAINER_HOST} -d xdebug.client_port=${PHP_XDEBUG_PORT} -d memory_limit=256M"
 fi
 
+if [ "${PHP_XDEBUG_MODE:-}" = "coverage" ]; then
+    XDEBUG_MODE="-e XDEBUG_MODE=coverage"
+fi
+
 # Suite execution
 case ${TEST_SUITE} in
     acceptance)
